vm_supervisor/storage.py

```diff
@@ -311,3 +317,37 @@ async def get_volume_path(volume: MachineVolume, namespace: str) -> Path:
             return volume_path
     else:
         raise NotImplementedError("Only immutable volumes are supported")
+
+
+async def create_volume_snapshot(path: Path) -> Path:
+    new_path = Path(f"{path}.{datetime.today().strftime('%d%m%Y-%H%M%S')}.bak")
+    copy2(path, new_path)
+    return new_path
+
+
+async def compress_volume_snapshot(
+    path: Path,
+    algorithm: SnapshotCompressionAlgorithm = SnapshotCompressionAlgorithm.gz,
+) -> Path:
+    if algorithm != SnapshotCompressionAlgorithm.gz:
+        raise NotImplementedError
+
+    new_path = Path(f"{path}.gz")
+
+    await run_in_subprocess(
+        [
+            str(algorithm),
+            str(path),
+        ]
+    )
```

Comment:
Okay so one problem you're gonna have here is that you're gonna end up with one process per instance. At the moment this may be acceptable, let's measure the overhead once this is deployed.